packages/containers/src/HeaderBar/HeaderBar.container.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import omit from 'lodash/omit';
+import { Map } from 'immutable';
+import { cmfConnect, Inject } from '@talend/react-cmf';
+
+import Constants from './HeaderBar.constant';
+
+export const DEFAULT_STATE = new Map({
+	fetchingProducts: false,
+});
+
+const HeaderBar = props => {
+	const { productsUrl, productsLang, productsItems, products = {}, ...componentProps } = props;
+
+	if (productsItems) {
+		// Add onClickDispatch event to items
+		products.items = productsItems.map(product => ({
+			onClickDispatch: { type: Constants.HEADER_BAR_OPEN_PRODUCT, payload: product },
+			...product,
+		}));
+
+		componentProps.products = products;
+	} else if (!props.state.get('fetchingProducts')) {
+		// Trigger fetch if not already fetching
+		props.dispatch({
```

Comment:
please never dispatch in a render function !

Reply:
What do you suggest for this ?
We couldn't make it work with that process in componentWillMount